builtin/providers/test/resource_with_custom_diff.go

```diff
@@ -0,0 +1,84 @@
+package test
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func testResourceCustomDiff() *schema.Resource {
+	return &schema.Resource{
+		Create:        testResourceCustomDiffCreate,
+		Read:          testResourceCustomDiffRead,
+		CustomizeDiff: testResourceCustomDiffCustomizeDiff,
```

Comment:
Hmm, so it turns out that this can't necessarily function because it collides with the already existing `resource.Diff`, of course... doh!

Trying to think of something else that still keeps the spirit of what we are trying to do here...